homeassistant/components/sensor/duke_energy.py

```diff
@@ -0,0 +1,81 @@
+"""
+Support for Duke Energy Gas and Electric meters.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/sensor/duke_energy/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pydukeenergy==0.0.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup all Duke Energy meters."""
+    from pydukeenergy.api import DukeEnergy, DukeEnergyException
+
+    try:
+        duke = DukeEnergy(config.get(CONF_USERNAME),
+                          config.get(CONF_PASSWORD),
+                          update_interval=500)
+    except DukeEnergyException:
+        _LOGGER.error("Failed to setup Duke Energy")
+        return False
+
+    meters = duke.get_meters()
+    sensors = []
+    for meter in meters:
+        sensors.append(DukeEnergyMeter(meter))
+    add_devices(sensors)
+
+    return True
```

Comment:
don't need to return anything (no one is checking)

Reply:
I thought you had to return False if setup failed? Is that not true anymore?